go/parquet/s3/s3_reader.go

```diff
@@ -0,0 +1,305 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package s3
+
+import (
+	"bytes"
+	"context"
+	"encoding/binary"
+	"fmt"
+	"io"
+	"runtime"
+	"sync"
+
+	"github.com/apache/arrow/go/v11/arrow/memory"
+	"github.com/apache/arrow/go/v11/parquet"
+	"github.com/apache/arrow/go/v11/parquet/internal/encryption"
+	"github.com/apache/arrow/go/v11/parquet/metadata"
+	"github.com/xitongsys/parquet-go-source/s3"
+	"github.com/xitongsys/parquet-go/source"
+	"golang.org/x/xerrors"
+)
+
+const (
+	footerSize uint32 = 8
+)
+
+var (
+	magicBytes                  = []byte("PAR1")
+	magicEBytes                 = []byte("PARE")
+	errInconsistentFileMetadata = xerrors.New("parquet: file is smaller than indicated metadata size")
+)
+
+type S3file struct {
+	source.ParquetFile
+}
+
+func (rdr S3file) ReadAt(p []byte, off int64) (n int, err error) {
+	rdr.Seek(off, io.SeekCurrent)
+	return rdr.Read(p)
+}
```

Comment:
You could like just pass a pointer to this to `file.OpenParquetFile` and it would work. The rest of this file is then unnecessary and just copying code.